mods/creative/inventory.lua

```diff
@@ -0,0 +1,181 @@
+	local player_name = player:get_player_name()
+	player_inventory[player_name] = {}
+	player_inventory[player_name].size = 0
+	player_inventory[player_name].filter = ""
+	player_inventory[player_name].start_i = 1
+
+	minetest.create_detached_inventory("creative_" .. player_name, {
+		allow_move = function(inv, from_list, from_index, to_list, to_index, count, player)
+			if creative_mode and not to_list == "main" then
+				return count
+			else
+				return 0
+			end
+		end,
+		allow_put = function(inv, listname, index, stack, player)
+			return 0
+		end,
+		allow_take = function(inv, listname, index, stack, player)
+			if creative_mode then
+				return -1
+			else
+				return 0
+			end
+		end,
+		on_move = function(inv, from_list, from_index, to_list, to_index, count, player)
+		end,
+		on_put = function(inv, listname, index, stack, player)
+		end,
+		on_take = function(inv, listname, index, stack, player)
+			local player_name, stack_name = player:get_player_name(), stack:get_name()
+			if stack then
+				minetest.log("action", player_name .. " takes " .. stack_name .. " from creative inventory")
+			end
+		end,
+	})
+
+	creative.update_creative_inventory(player_name, minetest.registered_items)
+end
+
+function creative.update_creative_inventory(player_name, tab_content)
+	local creative_list = {}
+	local player_inv = minetest.get_inventory({type = "detached", name = "creative_" .. player_name})
+	local inv = player_inventory[player_name]
+
+	for name, def in pairs(tab_content) do
+		if not (def.groups.not_in_creative_inventory == 1) and
+				def.description and def.description ~= "" and
+				(def.name:find(inv.filter, 1, true) or
+					def.description:lower():find(inv.filter, 1, true)) then
+			creative_list[#creative_list+1] = name
+		end
+	end
+
+	table.sort(creative_list)
+	player_inv:set_size("main", #creative_list)
+	player_inv:set_list("main", creative_list)
+	inv.size = #creative_list
+end
+
+-- Create the trash field
+local trash = minetest.create_detached_inventory("creative_trash", {
+	-- Allow the stack to be placed and remove it in on_put()
+	-- This allows the creative inventory to restore the stack
+	allow_put = function(inv, listname, index, stack, player)
+		if creative_mode then
+			return stack:get_count()
+		else
+			return 0
+		end
+	end,
+	on_put = function(inv, listname)
+		inv:set_list(listname, {})
+	end,
+})
+trash:set_size("main", 1)
+
+creative.formspec_add = ""
+
+function creative.register_tab(name, title, items)
+	sfinv.register_page("creative:" .. name, {
+		title = title,
+		get = function(self, player, context, vars)
+			local player_name = player:get_player_name()
+			creative.update_creative_inventory(player_name, items)
+			local inv = player_inventory[player_name]
+			local start_i = inv.start_i or 0
+			local pagenum = math.floor(start_i / (3*8) + 1)
+			local pagemax = math.ceil(inv.size / (3*8))
+			return [[
+					size[8,8.6]
+					bgcolor[#080808BB;true]
+					background[5,5;1,1;gui_formbg.png;true]
```

Comment:
Background is an image which is drawn below all over elements, this background image is at (5, 5) and has size (1, 1). I imagine that default.gui_bg covers the entire window